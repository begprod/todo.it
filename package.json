{
  "name": "todo.it",
  "version": "2.6.2",
  "description": "Another todo app but with features",
  "repository": "https://github.com/begprod/todo.it",
  "homepage": "https://begprod.github.io/todo.it/",
  "author": "Andrey Aratov",
  "license": "MIT",
  "private": true,
  "type": "module",
  "scripts": {
    "dev": "vite",
    "build": "vue-tsc --noEmit && vite build",
    "preview": "vite preview",
    "test:unit:watch": "vitest",
    "test:unit": "vitest run",
    "prettier:check": "prettier --check src/",
    "prettier:format": "prettier --write src/",
    "type:check": "vue-tsc --noEmit -p tsconfig.vitest.json --composite false",
    "lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore",
    "deploy": "gh-pages -d dist",
    "prepare": "husky install"
  },
  "dependencies": {
    "@heroicons/vue": "^2.0.18",
    "@vueuse/core": "^10.2.0",
    "axios": "^1.6.0",
    "date-fns": "^2.30.0",
    "markdown-it": "^13.0.2",
    "markdown-it-task-lists": "^2.1.1",
    "pinia": "^2.1.3",
    "uniqid": "^5.4.0",
    "vue": "^3.3.7",
    "vue-router": "^4.2.2",
    "vuedraggable": "^4.1.0"
  },
  "devDependencies": {
    "@pinia/testing": "^0.1.3",
    "@rushstack/eslint-patch": "^1.2.0",
    "@tsconfig/node18": "^2.0.1",
    "@types/jsdom": "^21.1.1",
    "@types/markdown-it": "^13.0.6",
    "@types/node": "^18.16.17",
    "@types/uniqid": "^5.3.2",
    "@vitejs/plugin-vue": "^4.2.3",
    "@vue/eslint-config-prettier": "^7.1.0",
    "@vue/eslint-config-typescript": "^11.0.3",
    "@vue/test-utils": "^2.4.1",
    "@vue/tsconfig": "^0.4.0",
    "autoprefixer": "^10.4.14",
    "eslint": "^8.39.0",
    "eslint-plugin-vue": "^9.11.0",
    "gh-pages": "^6.0.0",
    "husky": "^8.0.0",
    "jsdom": "^22.1.0",
    "npm-run-all": "^4.1.5",
    "postcss": "^8.4.31",
    "prettier": "^3.1.1",
    "sass": "^1.63.6",
    "tailwindcss": "^3.3.2",
    "typescript": "^5.2.2",
    "vite": "^5.0.12",
    "vite-plugin-pwa": "^0.17.4",
    "vitest": "^1.2.2",
    "vue-tsc": "^1.8.25"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
