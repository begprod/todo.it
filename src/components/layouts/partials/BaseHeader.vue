<template>
  <header class="fixed top-0 w-full bg-white z-20">
    <div class="max-w-base mx-auto p-5 flex items-center justify-between">
      <div class="flex items-center">
        <a href="/" class="flex items-end text-4xl font-bold leading-10">
          <div class="logo-gradient gradient-animation rounded-xl">todo.it</div>
          <sub class="logo-sub-gradient gradient-animation text-sm ml-2 bottom-0 rounded-xl">beta</sub>
        </a>
      </div>
      <a href="https://github.com/begprod/todo.it" target="_blank">
        <img src="@/assets/images/github-mark.svg" class="w-10 opacity-80 hover:opacity-100 transition-all duration-300" alt="todo.it github">
      </a>
    </div>
  </header>
</template>

<style scoped lang="scss">
.logo-gradient {
  background-color: #4158D0;
  background-image: linear-gradient(43deg, #4158D0 0%, #C850C0 46%, #FFCC70 100%);
  background-size: 400%;
  background-repeat: repeat;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  -moz-background-clip: text;
  -moz-text-fill-color: transparent;
}

.logo-sub-gradient {
  background-color: #4158D0;
  background-image: linear-gradient(43deg, #FFCC70 0%, #C850C0 46%, #4158D0 100%);
  background-size: 400%;
  background-repeat: repeat;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  -moz-background-clip: text;
  -moz-text-fill-color: transparent;
}

.gradient-animation {
  animation: gradient 15s ease-in-out infinite;
}

@keyframes gradient {
  0% {
    background-position: 0% 50%;
  }

  50% {
    background-position: 100% 50%;
  }

  100% {
    background-position: 0% 50%;
  }
}
</style>
