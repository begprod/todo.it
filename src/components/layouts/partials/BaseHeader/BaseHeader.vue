<template>
  <header class="sticky top-0 w-full bg-white z-[100]">
    <div class="container mx-auto p-3 flex items-center justify-between">
      <div class="flex items-center">
        <a href="/todo.it/" class="flex items-end font-bold">
          <div class="logo-gradient gradient-animation text-2xl">todo.it</div>
          <sub class="logo-sub-gradient gradient-animation text-sm ml-2 bottom-0">
            v{{ version }}
          </sub>
        </a>
      </div>
      <a href="https://github.com/begprod/todo.it" target="_blank">
        <img
          src="@/assets/images/github-mark.svg"
          class="w-7 opacity-80 hover:opacity-100 transition-all duration-300"
          width="40"
          height="40"
          alt="todo.it github"
        />
      </a>
    </div>
  </header>
</template>

<script setup lang="ts">
// @ts-ignore
import { version } from '../../../../../package.json';
</script>

<style scoped lang="scss">
.logo-gradient {
  background-color: #4158d0;
  background-image: linear-gradient(43deg, #4158d0 0%, #c850c0 46%, #ffcc70 100%);
  background-size: 400%;
  background-repeat: repeat;
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  -moz-background-clip: text;
  -moz-text-fill-color: transparent;
}

.logo-sub-gradient {
  background-color: #4158d0;
  background-image: linear-gradient(43deg, #ffcc70 0%, #c850c0 46%, #4158d0 100%);
  background-size: 400%;
  background-repeat: repeat;
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  -moz-background-clip: text;
  -moz-text-fill-color: transparent;
}

.gradient-animation {
  animation: gradient 15s ease-in-out infinite;
}

@keyframes gradient {
  0% {
    background-position: 0% 50%;
  }

  50% {
    background-position: 100% 50%;
  }

  100% {
    background-position: 0% 50%;
  }
}
</style>
