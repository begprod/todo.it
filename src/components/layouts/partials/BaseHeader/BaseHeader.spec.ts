import { version } from '../../../../../package.json';
import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import BaseHeader from '@/components/layouts/partials/BaseHeader/BaseHeader.vue';

describe('BaseHeader', () => {
  const wrapper = mount(BaseHeader);

  it('should contain css classes', () => {
    expect(wrapper.html()).toContain('sticky top-0 w-full bg-white z-[100]');
    expect(wrapper.html()).toContain('container mx-auto p-5 flex items-center justify-between');
    expect(wrapper.html()).toContain('logo-gradient gradient-animation text-2xl lg:text-4xl');
    expect(wrapper.html()).toContain('logo-sub-gradient gradient-animation text-sm ml-2 bottom-0');
    expect(wrapper.html()).toContain(
      'w-7 lg:w-10 opacity-80 hover:opacity-100 transition-all duration-300',
    );
  });

  it('should contain github image', () => {
    expect(wrapper.html()).toContain('/todo.it/src/assets/images/github-mark.svg');
  });

  it('should contain app version', () => {
    expect(wrapper.html()).toContain(`v${version}`);
  });
});
