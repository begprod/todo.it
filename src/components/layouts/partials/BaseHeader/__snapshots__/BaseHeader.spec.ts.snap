// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseHeader > should match snapshot BaseHeader 1`] = `
"<header data-v-d3c55262=\\"\\" class=\\"sticky top-0 w-full bg-white z-[100]\\">
  <div data-v-d3c55262=\\"\\" class=\\"container mx-auto p-5 flex items-center justify-between\\">
    <div data-v-d3c55262=\\"\\" class=\\"flex items-center\\"><a data-v-d3c55262=\\"\\" href=\\"/todo.it/\\" class=\\"flex items-end font-bold\\">
        <div data-v-d3c55262=\\"\\" class=\\"logo-gradient gradient-animation text-2xl lg:text-4xl\\">todo.it</div><sub data-v-d3c55262=\\"\\" class=\\"logo-sub-gradient gradient-animation text-sm ml-2 bottom-0\\"> v2.3.1</sub>
      </a></div><a data-v-d3c55262=\\"\\" href=\\"https://github.com/begprod/todo.it\\" target=\\"_blank\\"><img data-v-d3c55262=\\"\\" src=\\"/todo.it/src/assets/images/github-mark.svg\\" class=\\"w-7 lg:w-10 opacity-80 hover:opacity-100 transition-all duration-300\\" width=\\"40\\" height=\\"40\\" alt=\\"todo.it github\\"></a>
  </div>
</header>"
`;
